test/kubernetes_unittest.cc

```diff
@@ -717,41 +717,72 @@ bool WaitForNewerCollectionTimestamp(const MetadataStore& store,
   return false;
 }
 
-TEST_F(KubernetesTest, KubernetesUpdater) {
+void TestKubernetesUpdater(bool cluster_level_metadata, bool service_metadata) {
+  const std::string nodes_path = cluster_level_metadata ?
+    "/api/v1/watch/nodes/?watch=true" :
+    "/api/v1/watch/nodes/TestNodeName?watch=true";
+  const std::string pods_path = cluster_level_metadata ?
+    "/api/v1/pods?fieldSelector=spec.nodeName%3D&watch=true" :
+    "/api/v1/pods?fieldSelector=spec.nodeName%3DTestNodeName&watch=true";
+  // These 2 paths only used when both cluster_level_metadata and
+  // service_metadata are true.
+  const std::string services_path =
+    "/api/v1/watch/services/?watch=true";
+  const std::string endpoints_path =
+    "/api/v1/watch/endpoints/?watch=true";
+
   // Create a fake server representing the Kubernetes master.
   testing::FakeServer server;
   server.SetResponse("/api/v1/nodes?limit=1", "{}");
   server.SetResponse("/api/v1/pods?limit=1", "{}");
-  server.AllowStream(
-      "/api/v1/pods?fieldSelector=spec.nodeName%3DTestNodeName&watch=true");
-  server.AllowStream(
-      "/api/v1/watch/nodes/TestNodeName?watch=true");
+  server.AllowStream(nodes_path);
+  server.AllowStream(pods_path);
+  if (cluster_level_metadata && service_metadata) {
+    server.AllowStream(services_path);
+    server.AllowStream(endpoints_path);
+  }
 
   // Start the updater with a config that points to the fake server.
   Configuration config(std::istringstream(
       "InstanceId: TestID\n"
       "InstanceResourceType: gce_instance\n"
       "InstanceZone: TestZone\n"
+      "KubernetesClusterLevelMetadata: "
+      + std::string(cluster_level_metadata ? "true" : "false") + "\n"
```

Comment:
At some point, would it be helpful to have a bool_to_string method?

Reply:
Yeah, maybe there already exists some helper function?

Reply:
We do have `format::str(bool)`, but it's broken. I would be game to changing the implementation to
```c++
std::string str(bool b) { return b ? "true" : "false"; }
```
and using that here.

Reply:
No longer needed.